codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/awsjson/AWSJsonHttpResponseBindingErrorGenerator.kt

```diff
@@ -19,7 +19,43 @@ import software.amazon.smithy.swift.codegen.model.toUpperCamelCase
 import software.amazon.smithy.swift.codegen.utils.errorShapeName
 
 class AWSJsonHttpResponseBindingErrorGenerator : HttpResponseBindingErrorGeneratable {
-    override fun render(ctx: ProtocolGenerator.GenerationContext, op: OperationShape, unknownServiceErrorSymbol: Symbol) {
+    override fun renderServiceError(ctx: ProtocolGenerator.GenerationContext) {
+        val serviceShape = ctx.service
+        val serviceName = ctx.service.id.name
+        val rootNamespace = ctx.settings.moduleName
+
+        ctx.delegator.useFileWriter("./$rootNamespace/models/$serviceName+ServiceErrorHelperMethod.swift") { writer ->
+            writer.addImport(AWSSwiftDependency.AWS_CLIENT_RUNTIME.target)
+            writer.addImport(SwiftDependency.CLIENT_RUNTIME.target)
+
+            writer.openBlock("extension ${ctx.symbolProvider.toSymbol(ctx.service).name}Types {", "}") {
+                writer.openBlock(
+                    "static func makeServiceError(_ httpResponse: \$N, _ decoder: \$D, _ error: \$N, _ id: String?) async throws -> \$N? {", "}",
+                    ClientRuntimeTypes.Http.HttpResponse,
+                    ClientRuntimeTypes.Serde.ResponseDecoder,
+                    AWSClientRuntimeTypes.RestJSON.RestJSONError,
+                    SwiftTypes.Error
+                ) {
+                    writer.openBlock("switch error.errorType {", "}") {
+                        val serviceErrorShapes =
+                            serviceShape.errors.map { ctx.model.expectShape(it) as StructureShape }.toSet().sorted()
+                        for (errorShape in serviceErrorShapes) {
+                            val errorShapeName = errorShape.errorShapeName(ctx.symbolProvider)
+                            val errorShapeType = ctx.symbolProvider.toSymbol(errorShape).name
+                            writer.write(
+                                "case \$S: return try await \$L(httpResponse: httpResponse, decoder: decoder, message: error.errorMessage, requestID: id)",
+                                errorShapeName,
+                                errorShapeType
+                            )
+                        }
```

Comment:
```
serviceErrorShapes.forEach { errorShape ->
    val errorShapeName = errorShape.errorShapeName(ctx.symbolProvider)
    val errorShapeType = ctx.symbolProvider.toSymbol(errorShape).name
    write(
        "case \$S: return try await \$L(httpResponse: httpResponse, decoder: decoder, message: error.errorMessage, requestID: id)",
        errorShapeName,
        errorShapeType
    )
}
```

Reply:
Changed `for-in` loop to `.forEach { element -> . . .}` as suggested.